### PR TITLE
Enable checking format strings, suppress many warnings

### PR DESCRIPTION
**Pull request by edwardalee**

### CODE REVIEW

File: core/utils/util.h [hunk 1]
```diff
@@ -35,6 +35,18 @@ THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <stdarg.h>   // Defines va_list
 
+// To silence warnings about a function being a candidate for format checking
+// with gcc, add an attribute.
+// The arguments are the position of the format string (starting with 1)
+// and the start of the remaining arguments, or 0 for vprintf style functions.
+// NOTE: For some weird reason, the C compiler used in the Windows CI has
+// __GNUC__ defined but is not gcc. So we have to exclude Windows.
```
Comment (Soroosh129) [inline core/utils/util.h:43]: Is this note still relevant?

**Review by Soroosh129 — APPROVED** [review]
Looks great!

File: core/utils/util.h (hunk 1 above)
Comment (edwardalee) [reply core/utils/util.h]: Removed obsolete note.

**Review by petervdonovan — APPROVED** [review]
LGTM!

**Pull request merged**